### PR TITLE
[DC-1787] Create cleaning rule to remove records for pids not in CDR

### PR DESCRIPTION
**Pull request by nishanthpp93**
* [DC-1787] Create CR to remove data for PIDs not in CDR
* [DC-1787] Update CR and add test

[DC-1787]: https://precisionmedicineinitiative.atlassian.net/browse/DC-1787
[DC-1787]: https://precisionmedicineinitiative.atlassian.net/browse/DC-1787

### CODE REVIEW

File: data_steward/cdr_cleaner/cleaning_rules/remove_non_existing_pids.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+WHERE person_id NOT IN
+(SELECT person_id
+FROM `{{project_id}}.{{reference_dataset_id}}.person`)
+""")
+
+
+class RemoveNonExistingPids(BaseCleaningRule):
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id,
+                 reference_dataset_id):
```
Comment (jp3477) [inline data_steward/cdr_cleaner/cleaning_rules/remove_non_existing_pids.py:37]: How would clean_cdr know what reference_dataset_id to pass? Would it be worth making `reference_dataset_id` a keyword param?

**Review by jp3477 — COMMENTED** [review]
Mostly looks good, just left a question.

File: data_steward/cdr_cleaner/cleaning_rules/remove_non_existing_pids.py (hunk 1 above)
Comment (nishanthpp93) [reply data_steward/cdr_cleaner/cleaning_rules/remove_non_existing_pids.py:37]: Good question, that part is handled by the clean engine [here](https://github.com/all-of-us/curation/blob/242b53e1467baad726a4ff6a1c851f67c17e8be6/data_steward/cdr_cleaner/clean_cdr_engine.py#L169). Do you mean adding a default value? It ideally shouldn't need it, and you can see another example in [survey_version_info.py](https://github.com/all-of-us/curation/blob/develop/data_steward/cdr_cleaner/manual_cleaning_rules/survey_version_info.py). Not sure if that answers your question, let me know.

Comment (jp3477) [reply data_steward/cdr_cleaner/cleaning_rules/remove_non_existing_pids.py:37]: Ah I see, that makes sense. Thanks for sharing.

**Review by jp3477 — APPROVED** [review]
Nice job!

**Pull request merged**